curriculum/build-curriculum.js

```diff
@@ -0,0 +1,377 @@
+ * @param {string} params.engId - The English ID to look up in dictionaries
+ * @param {string} params.text - The fallback English text to use if translation not found
+ * @returns {Object} Object mapping language codes to translated text or fallback English text
+ */
+function getTranslationEntry(dicts, { engId, text }) {
+  return Object.keys(dicts).reduce((acc, lang) => {
+    const entry = dicts[lang][engId];
+    if (entry) {
+      return { ...acc, [lang]: entry };
+    } else {
+      // default to english
+      return { ...acc, [lang]: text };
+    }
+  }, {});
+}
+
+/**
+ * Creates a mapping of English comments to their translations across all supported languages
+ * @param {string} dictionariesDir - Path to the main (english) dictionaries directory
+ * @param {string} targetDictionariesDir - Path to the target (i18n or english) dictionaries directory
+ * @returns {Object} Object mapping English comment text to translations in all languages
+ */
+function createCommentMap(dictionariesDir, targetDictionariesDir) {
+  debug(
+    `Creating comment map from ${dictionariesDir} and ${targetDictionariesDir}`
+  );
+  const languages = fs.readdirSync(targetDictionariesDir);
+
+  const dictionaries = languages.reduce((acc, lang) => {
+    const commentsPath = path.resolve(
+      targetDictionariesDir,
+      lang,
+      'comments.json'
+    );
+    const commentsData = JSON.parse(fs.readFileSync(commentsPath, 'utf8'));
+    return {
+      ...acc,
+      [lang]: commentsData
+    };
+  }, {});
+
+  const COMMENTS_TO_TRANSLATE = JSON.parse(
+    fs.readFileSync(
+      path.resolve(dictionariesDir, 'english', 'comments.json'),
+      'utf8'
+    )
+  );
+
+  const COMMENTS_TO_NOT_TRANSLATE = JSON.parse(
+    fs.readFileSync(
+      path.resolve(
+        dictionariesDir,
+        'english',
+        'comments-to-not-translate.json'
+      ),
+      'utf8'
+    )
+  );
+
+  // map from english comment text to translations
+  const translatedCommentMap = Object.entries(COMMENTS_TO_TRANSLATE).reduce(
+    (acc, [id, text]) => {
+      return {
+        ...acc,
+        [text]: getTranslationEntry(dictionaries, { engId: id, text })
+      };
+    },
+    {}
+  );
+
+  // map from english comment text to itself
+  const untranslatableCommentMap = Object.values(
+    COMMENTS_TO_NOT_TRANSLATE
+  ).reduce((acc, text) => {
+    const englishEntry = languages.reduce(
+      (acc, lang) => ({
+        ...acc,
+        [lang]: text
+      }),
+      {}
+    );
+    return {
+      ...acc,
+      [text]: englishEntry
+    };
+  }, {});
+
+  const allComments = { ...translatedCommentMap, ...untranslatableCommentMap };
+
+  // the english entries need to be added here, because english is not in
+  // languages
+  Object.keys(allComments).forEach(comment => {
+    allComments[comment].english = comment;
+  });
+
+  return allComments;
+}
+
+// Map of superblock filenames to their SuperBlocks enum values
+const superBlockNames = {
```

Comment:
I do not know how you want this to go down, but I assume we need to change, and re-use `shared/config/curriculum.ts > superBlockToFolderMap` instead of this.

Reply:
Yeah, I think `superBlockToFolderMap` can be deleted in favour of calling a curriculum function to get whatever is needed. Exposing the internals of the curriculum like this is a bad practice that I'd like to get rid of.

I plan to make these kinds of changes in the PR where I fix the challenge-helper-scripts, though. I've made a start and should have that done today/early next week.